crates/graphql_lang_types/src/graphql_type_annotation.rs

```diff
@@ -9,6 +9,25 @@ pub enum GraphQLTypeAnnotation<TValue> {
     NonNull(Box<GraphQLNonNullTypeAnnotation<TValue>>),
 }
 
+impl<TValue> From<GraphQLTypeAnnotation<TValue>> for GraphQLNonNullTypeAnnotation<TValue> {
+    fn from(value: GraphQLTypeAnnotation<TValue>) -> Self {
+        match value {
+            GraphQLTypeAnnotation::Named(named) => GraphQLNonNullTypeAnnotation::Named(named),
```

Comment:
This converting `Foo` to `Foo!`... if we need this, we should not use `From` for this, since it will be easy to unintentionally do this

Reply:
I want to avoid copying here, I could use something like this instead, do you have any better ideas?
```rust
    pub fn fold_non_null<OnNamed, OnList, TNewValue>(
        &self,
        on_named: OnNamed,
        on_list: OnList,
    ) -> TNewValue
    where
        OnNamed: FnOnce(&GraphQLNamedTypeAnnotation<TValue>) -> TNewValue,
        OnList: FnOnce(&GraphQLListTypeAnnotation<TValue>) -> TNewValue,
    {
        match self {
            GraphQLTypeAnnotation::Named(named) => on_named(named),
            GraphQLTypeAnnotation::List(list) => on_list(list),
            GraphQLTypeAnnotation::NonNull(non_null) => match non_null {
                GraphQLNonNullTypeAnnotation::Named(named) => on_named(named),
                GraphQLNonNullTypeAnnotation::List(list) => on_list(list),
            },
        }
    }
    ```

Reply:
With this I'd copy only in the case I encounter two `non_null` types because the function takes two nullable types, and the only way I can think of solving this is to make a new enum
```rust
enum GraphQLMaybeNullTypeAnnotation<T>{
  NonNull(GraphQLNonNullTypeAnnotation<T>),
  Null(GraphQLTypeAnnotation<T>)
}
```

Reply:
I'll think about this the next time I review this... which will most likely be in 2025 🎉, for now I need to pass out!

Though it's not a huge deal if we have to do some cloning here, these are really small objects